### ISSUE TITLE
Object placement and regional particularities questions.

### ISSUE THREAD

**Issue by genaray** [mapbox/mapbox-unity-sdk#736]
Dear Mapbox team,

im developing an little multiplayer game in mapbox. Server side i generate dozens of objects which will be placed somewhere near the player once he logs in and the map is loaded. So on the client side i receive multiple arrays of those objects. As next i loop them, extract the server side generated lat/lng position, spawn the certain gameobject in and finally calculate the unity position based on the lat/lng position using VectorExtensions.asUnityPosition(); . Well it works fine, but the problem is that its very resource intensive for the client to calculate those positions everytime he "jumps" to another spot or just loads the game at all. Im already using multiple threads for this process but its still noticeable.

How would you guys do something like that ? There arent any examples for such a case at all... So some help would be great.

Otherwise i want to generate region specific resources for the player ( server side ) once he is located in africa for example. Or also it would be great to spawn certain monsters in (server side ) based on the local temperate or time in the certain region .... But how would i do something like that ? 

Thanks for your great plugin ! Its awesome !

**Comment by jordy-isaac** [comment]
Hi @genaray! Happy to help. There are some general things you can do to better performance for your use case:

1. Adjust the map zoom level to reduce the number of objects displayed in one area.
2. Adjust the Camera Bounds settings in the Extent Options or the unity camera zoom to control the amount of objects shown on a screen.
3. Make sure that there is no colliders on the terrain / generated geometry. This has major performance implications.
4. if possible, uncheck the options for retina resolution and mip maps.

I would check out the DataExplorer documentation to see all of the settings available here: https://www.mapbox.com/unity-sdk/overview/data-explorer/#custom-data

For implementation of conditional objects based on weather or time, something like querying the GPS lat / lon from the users device and using a weather or time api to get the info at runtime would be a solution worth exploring. 

I hope this helps.

**Comment by genaray** [comment]
@jordy-isaac Thanks for the help ! The tipp with the performance already was great ! 

So do i undertstand that right ? Mapbox doesnt support "GeoCoding" or how ever this is called ? Do i really need to use the google maps api for acessing those values based on the location ?

Thanks again :)

**Comment by atripathi-mb** [comment]
@genaray 
Few comments - 
 1. Have you looked into using Mapbox custom tilesets? You can same the location specific properties as a point feature with attributes and retrieve them using the mapbox unity sdk pipeline. 
2. Mapbox has a geocoding/reverse geocoding api available ( address to gps and vice -versa) but this does not provide temperature or time information. 

@jordy-isaac already suggested performance best practices. 
If you can give us some more details on what you are trying to achieve, we can suggest some more options.

**Comment by genaray** [comment]
@atripathi-mb @jordy-isaac  Yeah i already took a look at the custom tilesets... but thats not what i want ^^

So my whole game is server based. When a player enters a new self calculated chunk, it sends his lat/lng position to my server. The server processes this request, generates "objects" near the player within the same tile and sends this array of objects back to the client. I need to check serverside what time and temperature exists near the players lat/lng position. Futhermore i need to check in which continent the given lat/lng position lays in. Depending on this the sever will generate different "objects". 

If mapbox doesnt support this kind of geocoding i would need to use google maps api... thats the only way i found till now to solve this iusse.

**Comment by atripathi-mb** [comment]
@genaray 
> thats the only way i found till now to solve this iusse.
Could you point to the specific API's you are referring to ? Or what data do you plan to use to get the which tree to spawn? That way I can point you to the right API/data ?

**Comment by genaray** [comment]
@atripathi-mb @jordy-isaac Well as i Said multiple Times above i only need the temperature, local time and Continent Data of the certain lat/lng.

The specific API im referring to is the Google Maps API ... Their geocoding allowes me to receive lat/lng specific Data ( Country, temperature, time, City it May belong to etc ... ) By using url queries .

**Comment by atripathi-mb** [comment]
@genaray 
As far as I know,  Google's geocoding API will not return temparature or time (I could be wrong) . 

Here is what Mapbox geocoding provides you https://www.mapbox.com/api-documentation/#response-object which is equivalent to Google Maps geocoding API at the quick comparison that I did. 

I asked you the specific field in the geocode response that you were using to get temperature & time since I am not aware of it.

**Comment by genaray** [comment]
@atripathi-mb Oh im sorry... i exchanged something... Of course the google maps api doesnt return temperature and time... They once did... but they shutted this part of the service down. Instead there are several weather apis. But google maps timezones api still returns the timezone and its possible to calculate the certain time of a location with those variables.

**Comment by atripathi-mb** [comment]
@genaray you can use this repo https://github.com/mapbox/timespace to do timezone queries after your geocoding query.

**Comment by genaray** [comment]
@atripathi-mb Well thanks a lot :) Is there also a java api for the mapbox timespace library ?
